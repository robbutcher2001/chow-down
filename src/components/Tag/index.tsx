import React, { FunctionComponent } from 'react';

import styled from 'styled-components';

interface TagProps {
  label: string,
  $colour?: string
};

interface StyledTagProps {
  $colour?: string
}

const StyledTag = styled.span<StyledTagProps>`
  color: #fff;
  background-color: ${props => props.$colour};
  margin: 1px;
  padding: 0 0.5rem;
  font-size: 12px;
  border: 2px solid transparent;
  border-radius: 2em;
`

const Tag: FunctionComponent<TagProps> = (props: TagProps) => {

  return (
    <StyledTag $colour={props.$colour} >
      {props.label}
    </StyledTag>
  );
};

export default Tag;