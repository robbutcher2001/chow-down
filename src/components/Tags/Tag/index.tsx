import React, { FunctionComponent, ReactNode } from 'react';

import styled from 'styled-components';

interface TagProps {
  children: ReactNode;
  $colour?: string;
};

interface StyledTagProps {
  readonly $colour?: string;
}

const StyledTag = styled.span<StyledTagProps>`
  color: ${props => props.theme.colour.white};
  background-color: ${props => props.$colour};
  padding: 0 0.5rem;
  font-size: ${props => props.theme.typography.fontSize.xsmall};
  border: 2px solid transparent;
  border-radius: 5px;
`

const Tag: FunctionComponent<TagProps> = (props: TagProps) =>
  <StyledTag $colour={props.$colour}>
    {props.children}
  </StyledTag>;

export default Tag;