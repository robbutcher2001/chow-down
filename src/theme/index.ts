import { DefaultTheme } from 'styled-components';

import colour from './defs/_colour';

const theme: DefaultTheme = {
  colour
};

export default theme;