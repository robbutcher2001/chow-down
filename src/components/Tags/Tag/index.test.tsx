import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from '../../../themes';
import Tag from '.';

test('Tag basic snapshot render', () => {
  const tag = renderer.create(
    <ThemeProvider theme={theme}>
      <Tag>
        <div>a child node</div>
      </Tag>
    </ThemeProvider>
  );

  expect(tag.toJSON()).toMatchSnapshot();
});

test('Tag correct tag content assertion', () => {
  const { getByText } = render(
    <ThemeProvider theme={theme}>
      <Tag>
        <div>a child node</div>
      </Tag>
    </ThemeProvider>
  );

  expect(getByText(/a child node/)).toBeTruthy();
});