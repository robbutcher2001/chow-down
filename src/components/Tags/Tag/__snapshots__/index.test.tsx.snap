// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag basic snapshot render 1`] = `
<span
  className="sc-AxjAm hJJxwF"
>
  <div>
    a child node
  </div>
</span>
`;
